### PR TITLE
vdk-structlog: add syslog handler

### PR DESCRIPTION
**Pull request by duyguHsnHsn**
Stacked on top of https://github.com/vmware/versatile-data-kit/pull/3002

Solving: https://github.com/vmware/versatile-data-kit/issues/2953 

Tests: Unit tests for the function adding the configurations

Signed-off-by: Duygu Hasan [hduygu@vmware.com](mailto:hduygu@vmware.com)

### CODE REVIEW

File: projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py [hunk 1]
```diff
@@ -177,10 +205,22 @@ def initialize_job(self, context: JobContext) -> None:
...
+        syslog_handler = configure_syslog_handler(syslog_enabled, syslog_host, syslog_port,
+                                                  syslog_protocol, job_name, attempt_id)
+        if syslog_handler:
+            root_logger.addHandler(syslog_handler)
+
         out: HookCallResult
         out = yield
 
         root_logger.removeHandler(handler)
+        if syslog_handler:
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py:222]: We shouldn't be removing the syslog handler, we need it til the end of the job run. I had a problem previously, where adding a handler in this hook didn't result in having the handler in the next hook. That's why we remove it here and add it again in the `run_job` hook.

So we should either not remove the handler here or remove it here and add it again in the next hook. This should be tested in debug mode.

File: projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/constants.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2021-2024 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
 import logging
+import socket
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/constants.py:4]: We should merge this first https://github.com/vmware/versatile-data-kit/pull/3002

File: projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/syslog_config.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    formatter = logging.Formatter(DETAILED_FORMAT)
+    syslog_handler.setFormatter(formatter)
+
+    job_name_adder = AttributeAdder("job_name", job_name)
+    attempt_id_adder = AttributeAdder("attempt_id", attempt_id)
+
+    syslog_handler.addFilter(job_name_adder)
+    syslog_handler.addFilter(attempt_id_adder)
+
+    syslog_handler.setLevel("DEBUG")
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/syslog_config.py:51]: Was this set to DEBUG in core as well?

Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/syslog_config.py:51]: @DeltaMichael yes

File: projects/vdk-plugins/vdk-structlog/tests/test_structlog.py [hunk 4]
```diff
@@ -108,6 +110,44 @@ def test_structlog(log_format):
...
+            server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            server_socket.bind((host, port))
+            print(f"Syslog server listening on {host}:{port}")
+
+            while True:
+                data, addr = server_socket.recvfrom(1024)
+                syslog_out.append(f"{data.decode('utf-8')}")
+
+        server = threading.Thread(
+            target=start_syslog_server, args=("127.0.0.1", 8888), daemon=True
```
Comment (antoniivanov) [inline projects/vdk-plugins/vdk-structlog/tests/test_structlog.py:138]: I would pick less commonly used port.

**Pull request merged**